### PR TITLE
Implement remaining task combinators: and, andUntilDone

### PR DESCRIPTION
**Pull request by shadaj**

### CODE REVIEW

File: core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/AndUntilDoneTask.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  * A finite task where a continuous task is run while a finite task is running
+ *
+  * @param first the finite task to run
+  * @param second the continuous task to run at the same time
+  */
+class AndUntilDoneTask private[tasks](first: FiniteTask, second: ContinuousTask)
+  extends FiniteTask with FiniteTaskFinishedListener {
+  override def onFinished(task: FiniteTask): Unit = {
+    if (!first.isRunning) {
+      finished()
```
Comment (PhilipAxelrod) [inline core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/AndUntilDoneTask.scala:13]: Won't this cause endless recursion? Is that the point, it keeps going until first finishes?

File: core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/ParallelFiniteTask.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+  * A finite task where two subtasks are run in parallel
+  * @param first one of the tasks to run
+  * @param second the other task to run
+  */
+class ParallelFiniteTask private[tasks] (first: FiniteTask, second: FiniteTask)
+  extends FiniteTask with FiniteTaskFinishedListener {
+  override def onFinished(task: FiniteTask): Unit = {
+    if (!first.isRunning && !second.isRunning) {
```
Comment (PhilipAxelrod) [inline core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/ParallelFiniteTask.scala:11]: Same question as above

File: core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/AndUntilDoneTask.scala (hunk 1 above)
Comment (shadaj) [reply core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/AndUntilDoneTask.scala:13]: This is called when a subtask is finished, not when we are finished.

File: core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/ParallelFiniteTask.scala (hunk 2 above)
Comment (shadaj) [reply core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/ParallelFiniteTask.scala:11]: See above.

**Review by dtjong — APPROVED** [review]

**Pull request merged**